tests/unit_tests/test_kmers.py

```diff
@@ -0,0 +1,260 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+COPYRIGHT
+Copyright 2020 Ian J. Miller, Evan R. Rees, Kyle Wolf, Siddharth Uppal,
+Shaurya Chanana, Izaak Miller, Jason C. Kwan
+
+This file is part of Autometa.
+
+Autometa is free software: you can redistribute it and/or modify
+it under the terms of the GNU Affero General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+Autometa is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License
+along with Autometa. If not, see <http://www.gnu.org/licenses/>.
+COPYRIGHT
+
+Count, normalize and embed k-mers given nucleotide sequences
+"""
+
+import pytest
+import os
+
+import pandas as pd
+from unittest.mock import patch, MagicMock
+
+from autometa.common import kmers
+from autometa.common.exceptions import TableFormatError
+
+assembly = os.path.join("tests", "data", "metagenome.fna")
+test_df = pd.DataFrame(
+    {
+        "contig": [
+            "NODE_1_length_1389215_cov_225.275",
+            "NODE_2_length_1166739_cov_224.155",
+            "NODE_3_length_1063064_cov_225.095",
+            "NODE_4_length_1031470_cov_223.812",
+            "NODE_5_length_937195_cov_225.122",
+        ],
+        "AAAAA": [170, 1688, 143, 347, 156],
+        "AAAAG": [412, 1354, 345, 331, 317],
+        "AAATG": [536, 1513, 286, 325, 378],
+    }
+)
+test_df.set_index("contig", inplace=True)
+
+
+def test_revcomp():
+    string = "AAATTGCGCCCCCG"
+    rev_string = kmers._revcomp(string)
+    assert rev_string == "CGGGGGCGCAATTT"
+    assert type(rev_string) is str
+    assert len(rev_string) == 14
+
+
+def test_init_kmers():
+    unique_kmers_5 = kmers.init_kmers(kmer_size=5)
+    assert type(unique_kmers_5) is dict
+    assert len(unique_kmers_5) == 512
+    unique_kmers_3 = kmers.init_kmers(kmer_size=3.0)
+    assert type(unique_kmers_3) is dict
+    assert len(unique_kmers_3) == 32
+    unique_kmers_5 = kmers.init_kmers(kmer_size="5")
+    assert type(unique_kmers_5) is dict
+    assert len(unique_kmers_5) == 512
+    unique_kmers_3 = kmers.init_kmers(kmer_size="3.0")
+    assert type(unique_kmers_3) is dict
+    assert len(unique_kmers_3) == 32
+    with pytest.raises(TypeError):
+        kmers.init_kmers(kmer_size=5.5)
+    with pytest.raises(TypeError):
+        kmers.init_kmers(kmer_size="5.5")
```

Comment:
I do not think we should do any type conversion (in kmers.py) so we will not need multiple test cases here.